src/store/updater.rs

```diff
@@ -512,93 +518,109 @@ impl<'a> Updater<'a> {
     // TODO: look into hadoop-style reducers here
     /// Writes the state of the updater to the store, computing the Merkle tree updates along the
     /// way.
-    pub fn commit(self) -> Digest {
-        let Updater {
-            current_state,
-            store,
-        } = self;
+    pub fn commit(
+        self,
+    ) -> Result<
+        Digest,
+        TransactionError<
+            <S as TrieTransactional>::ErrorCreatingTransaction,
+            <<S as TrieTransactional>::Transaction as TrieWriter>::Error,
+        >,
+    >
+    where
+        S: TrieTransactional,
+    {
+        let mut current_state = self.current_state;
+
+        self.store.transaction(move |trie_writer| {
+            // Get ownership of the current state
+            let current_state = {
+                let mut my_current_state = UpdatingTrie::Empty;
+                std::mem::swap(&mut my_current_state, &mut current_state);
+                my_current_state
+            };
```

Comment:
```suggestion
            let current_state = std::mem::replace(&mut current_state, UpdatingTrie::Empty);
```

(That should be equivalent, I think?)

Reply:
Trying to omit it gives 

```
error[E0507]: cannot move out of `current_state.0`, as `current_state` is a captured variable in an `FnMut` closure
```

Again, @dwerner and I will have to revisit this when we figure out the right abstraction for transactions.

I've applied your suggestion in: a8be0d7e5c3c753fc1f95a190925bcf8e157a03e